File: backend/catalog-service/src/main/java/me/doruk/catalogservice/entity/Venue.java
package me.doruk.catalogservice.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "venue")
public class Venue {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id")
  private Long id;

  @Column(name = "name")
  @NotBlank(message = "Name must not be blank")
  @Size(max = 255, message = "Name must not exceed 255 characters")
  private String name;

  @Column(name = "location")
  @NotBlank(message = "Location must not be blank")
  @Size(max = 255, message = "Location must not exceed 255 characters")
  private String location;

  @Column(name = "total_capacity")
  @Min(value = 1, message = "Total capacity must be at least 1")
  private int totalCapacity;

  @Column(name = "image_url")
  @Size(max = 512, message = "Image URL must not exceed 512 characters")
  private String imageUrl;
}

File: backend/catalog-service/Entities/Venue.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CatalogService.Entities
{
    [Table("venue")]
    public class Venue
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public long Id { get; set; }

        [Column("name")]
        [Required(ErrorMessage = "Name must not be blank")]
        [StringLength(255, ErrorMessage = "Name must not exceed 255 characters")]
        public string? Name { get; set; } // Nullable

        [Column("location")]
        [Required(ErrorMessage = "Location must not be blank")]
        [StringLength(255, ErrorMessage = "Location must not exceed 255 characters")]
        public string? Location { get; set; }

        [Column("total_capacity")]
        [Range(1, int.MaxValue, ErrorMessage = "Total capacity must be at least 1")]
        public int TotalCapacity { get; set; }

        [Column("image_url")]
        [StringLength(512, ErrorMessage = "Image URL must not exceed 512 characters")]
        public string? ImageUrl { get; set; }

        public Venue() { }

        public Venue(long id, string name, string location, int totalCapacity, string imageUrl)
        {
            Id = id;
            Name = name;
            Location = location;
            TotalCapacity = totalCapacity;
            ImageUrl = imageUrl;
        }
    }
}
